arangod/Aql/InputAqlItemRow.cpp

```diff
@@ -213,6 +213,7 @@ bool InputAqlItemRow::isFirstDataRowInBlock() const noexcept {
   auto [shadowRowsBegin, shadowRowsEnd] = block().getShadowRowIndexesFrom(0);
 
   // Count the number of shadow rows before this row.
+  // cppcheck-suppress selfAssignment
   size_t const numShadowRowsBeforeCurrentRow = [this, shadowRowsBegin = shadowRowsBegin]() {
```

Comment:
I fixed this differently in https://github.com/arangodb/arangodb/pull/13921, without a suppression but with removing the IIFE. I prefer my solution as it is likely more obvious to the reader of the code.